common/frame.go

```diff
@@ -758,11 +758,13 @@ func (f *Frame) EvaluateHandle(pageFunc goja.Value, args ...goja.Value) (handle
 	}
 	f.executionContextMu.RUnlock()
 	if err != nil {
-		k6ext.Panic(f.ctx, "evaluating handle: %w", err)
+		// EvalHandle has a rich error, nolinting.
+		return nil, err //nolint:wrapcheck
```

Comment:
IMO we should wrap this error with the method name to give the error more context here, instead of wrapping it in `EvalHandle`.

If another method calls `EvalHandle` then `evaluating handle:` might be the wrong context for that caller.

Reply:
You mean something like `EvaluateHandle: evaluating handle:`?

Reply:
I agree that `EvaluateHandle: evaluating handle:` is not a good approach.

This is very minor so, don't worry about changing the code if it doesn't make sense. What i meant was that instead of wrapping the error in `EvalHandle` like so:

```go
return nil, fmt.Errorf("evaluating handle: %w", err)
```

why not just return `err` and allow this method (`EvaluateHandle`) to wrap the error with:

```go
 fmt.Errorf("evaluating handle: %w", err)
```

Another example is, in `EvalHandle` when `res` is `nil` it is currently wrapped like so:

```go
return nil, fmt.Errorf("evaluating handle: result is nil")
```

When it could be:

```go
return nil, fmt.Errorf("result is nil")
```

And in `EvaluateHandle` we would wrap the `err` with:

```go
fmt.Errorf("evaluating handle: %w", err).
```

Like i said, super minor so don't worry about this 🙂

Reply:
@ankur22, oh, I see what you mean now 🙂 Thanks for explaining 🙇

I couldn't see a difference in adding `"evaluating handle"` in `EvalHandle` vs. in`Page|Frame|JSHandle (ElementHandle)` since the final error message would be the same. So I've decided to go this way: [f040f90](https://github.com/grafana/xk6-browser/pull/826/commits/f040f9019edaeca5e017cbe577742d1059cee3de) to differentiate the error messages depending on where `EvalHandle` is called. Please check it out and let me know; thanks!

Reply:
🎉 Nice! That's exactly what i was thinking. Thanks 🙂